pkg/controller/windowsmachineconfig/nodeconfig/nodeconfig.go

```diff
@@ -324,6 +337,26 @@ func (nc *nodeConfig) waitForNodeAnnotation(annotation string) error {
 	return nil
 }
 
+// configureCNI populates the CNI config template and
+func (nc *nodeConfig) configureCNI() error {
+	// set the hostSubnet value in the clusterServiceCIDR struct
+	nc.nodeLevelConfig.SetHostSubnet(nc.node.Annotations[HybridOverlaySubnet])
+	// populate the CNI config file with the host subnet and the service network CIDR
+	configFile, err := nc.nodeLevelConfig.PopulateCniConfig(nc.clusterServiceCIDR)
+	if err != nil {
+		return errors.Wrapf(err, "error populating CNI config file %s", configFile)
+	}
+	if configFile == "" {
+		return errors.Wrapf(err, "received empty CNI configFile name")
+	}
+	// configure CNI in the Windows VM
+	if err = nc.Windows.ConfigureCNI(configFile); err != nil {
+		return errors.Wrapf(err, "error configuring CNI for %s", nc.node.GetName())
+	}
+	defer os.RemoveAll(configFile)
```

Comment:
This should be called when the configFile is created, not at the end of the function.